src/multi_physics/QED/src/physics/breit_wheeler/breit_wheeler_engine_tables_generator.hpp

```diff
@@ -0,0 +1,232 @@
+// 2) T.Erber Rev. Mod. Phys. 38, 626 (1966)
+// 3) C.P.Ridgers et al. Journal of Computational Physics 260, 1 (2014)
+// 4) A.Gonoskov et al. Phys. Rev. E 92, 023305 (2015)
+
+//Should be included by all the src files of the library
+#include "../../qed_commons.h"
+
+//Implements methods of BW lookup tables
+#include "breit_wheeler_engine_tables.hpp"
+//Uses BW tabulated functions
+#include "breit_wheeler_engine_tabulated_functions.hpp"
+//Uses progress bar
+#include "../../math/cmath_overloads.hpp"
+//Uses progress bar
+#include "../../utils/progress_bar.hpp"
+
+#include <omp.h>
+#include <vector>
+#include <chrono>
+#include <iostream>
+#include <type_traits>
+#include <stdexcept>
+
+namespace picsar{
+namespace multi_physics{
+namespace phys{
+namespace breit_wheeler{
+
+    //________________ dN/dt table _____________________________________________
+
+    /**
+    * Auxiliary function used to compute the T function in double precision
+    * with a single precision argument and to cast back the result to single precision
+    *
+    * @tparam RealType the floating point type to be used
+    * @tparam VectorType the vector type to be used (relevant for the class of which is a method is member)
+    *
+    * @param[in] x the value to be passed to compute_T_function
+    *
+    * @return the result of compute_T_function
+    */
+    template<typename RealType, typename VectorType>
+    PXRMP_INTERNAL_FORCE_INLINE_DECORATOR
+    RealType dndt_lookup_table<RealType, VectorType>::
+    aux_generate_double(RealType x)
+    {
+        return static_cast<RealType>(
+            compute_T_function<double>(x));
+    }
+
+    /**
+    * Generates the lookup table (not usable on GPUs).
+    *
+    * @tparam RealType the floating point type to be used
+    * @tparam VectorType the vector type to be used (relevant for the class of which is a method is member)
+    * @tparam Policy if set to generation_policy::force_internal_double it forces internal calculations in double precision
+    *
+    * @param[in] show_progress if true it shows a nice progress bar
+    */
+    template<typename RealType, typename VectorType>
+    template<generation_policy Policy>
+    void dndt_lookup_table<RealType, VectorType>::generate(
+        const bool show_progress)
+    {
+        constexpr bool use_internal_double =
+            (Policy == generation_policy::force_internal_double) &&
+            !std::is_same<RealType,double>();
+
+        auto t_start =  std::chrono::system_clock::now();
+
+        const auto all_coords = get_all_coordinates();
+        auto all_vals = std::vector<RealType>(all_coords.size());
+
+        int count = 0;
+        #pragma omp parallel for
+        for (int i = 0; i < all_coords.size(); ++i){
+            PXRMP_INTERNAL_CONSTEXPR_IF (use_internal_double){
+                all_vals[i] = aux_generate_double(all_coords[i]);
+            }
+            else {
+                all_vals[i] = compute_T_function(all_coords[i]);
+            }
+
```

Comment:
Maybe add a condition `if (show_progress)` here? `show_progress` seems to be currently unused (it's the same for the pair production table)